dashboard/test/controllers/api/v1/sections_controller_test.rb

```diff
@@ -523,6 +523,159 @@ class Api::V1::SectionsControllerTest < ActionController::TestCase
+    section_with_script = Section.find(section_with_script.id)
+
+    assert_equal(@course.id, section_with_script.course_id)
+    assert_nil(section_with_script.script_id)
+    assert_equal("My Section", section_with_script.name)
+    assert_equal(Section::LOGIN_TYPE_PICTURE, section_with_script.login_type)
+    assert_equal("K", section_with_script.grade)
+    assert_equal(false, section_with_script.stage_extras)
+    assert_equal(true, section_with_script.pairing_allowed)
+    assert_equal(false, section_with_script.hidden)
+  end
+
+  test "update: course_id is not updated if invalid" do
+    Course.stubs(:valid_course_id?).returns(false)
+
+    sign_in @teacher
+    section = create(:section, user: @teacher, course_id: nil)
+
+    post :update, params: {
+      id: section.id,
+      course_id: 1,
+    }
+    section.reload
+    assert_response :success
+    assert_nil section.course_id
+  end
+
+  test "update: script_id is not updated if invalid" do
+    Script.stubs(:valid_script_id?).returns(false)
+
+    sign_in @teacher
+    section = create(:section, user: @teacher, script_id: nil)
+
+    post :update, params: {
+      id: section.id,
+      script_id: 1,
+    }
+    section.reload
+    assert_response :success
+    assert_nil section.script_id
+  end
+
+  test "update: cannot update section you dont own" do
+    other_teacher = create(:teacher)
+    sign_in other_teacher
+    post :update, params: {
+      id: @section.id,
+      course_id: @course.id,
+    }
+    assert_response :forbidden
+  end
+
+  test "update: cannot update section if not logged in " do
+    post :update, params: {
+      id: @section.id,
+      course_id: @course.id,
+    }
+    assert_response :forbidden
+  end
+
+  test "update: can set course and script" do
+    sign_in @teacher
+    section = create(:section, user: @teacher, script_id: Script.flappy_script.id)
+    post :update, as: :json, params: {
+      id: section.id,
+      course_id: @csp_course.id,
+      script_id: @csp_script.id
+    }
+    assert_response :success
+    section.reload
+    assert_equal(@csp_course.id, section.course_id)
+    assert_equal(@csp_script.id, section.script_id)
+  end
+
+  test "update: non-matching course/script rejected" do
+    sign_in @teacher
+    section = create(:section, user: @teacher, script_id: Script.flappy_script.id)
+    post :update, params: {
+      id: section.id,
+      course_id: @course.id,
+      script_id: Script.artist_script.id
+    }
+    assert_response 400
+  end
+
+  test "update: can set course-less script" do
+    sign_in @teacher
+    section = create(:section, user: @teacher, script_id: Script.flappy_script.id)
+    post :update, params: {
+      id: section.id,
+      script_id: Script.artist_script.id
+    }
+    assert_response :success
+    section.reload
+    assert_nil section.course_id
+    assert_equal(Script.artist_script.id, section.script_id)
+  end
+
+  test "upate: setting a script results in UserScripts for students" do
```

Comment:
`s/upate/update`

Reply:
i'm sorry, i don't know what you mean by that

Reply:
Sorry; it's a [vim-ism for 'search-and-replace' or 'substitute'](http://vim.wikia.com/wiki/Search_and_replace). I meant to say there's a spelling error here, please replace "upate" with "update."

Reply:
oh! haha thank you, updating now